### PR TITLE
new login nav

### PR DESCRIPTION
**Pull request by hanbyul-here**
- new nav reflecint user's login state
- Since nav menus are not going to have straight href, `active-nav-link.js` to `main-nav.js`. `main-nav.js` does two things, 1.detecting user is logged in or not, manipulates dom. 2. detecting which tab should be active. It will be better to separate 2 function from 1 for future. 
- No affects until DOM part changes. 
- The example with new header is in `samples` folder (`blog-main-w-new-header.html`)

### CODE REVIEW

File: src/scripts/main-nav.js [hunk 1]
```diff
@@ -0,0 +1,235 @@
...
+    return 'sign out';
+  }
+
+  function putActiveTab () {
+    var navItems = document.querySelectorAll('.navbar-nav>li');
+    // After dom manipulation
+    var loginButton = document.getElementById('sign-in');
+    var signupButton = document.getElementById('sign-up');
+
+    if (isThisDevPortalPage()) {
```
Comment (sleepylemur) [inline src/scripts/main-nav.js]: if we're still using the old html, when we get here loginButton and signupButton will still be null, which will cause `loginButton.parentNode` to throw `Uncaught TypeError: Cannot read property 'parentNode' of null`.
can we update to `loginButton && isThisDevPortalPage()` here? or something equivalent?

**Comment by sleepylemur** [comment]
@hanbyul-here checked against current html, and it looks ok visually, but throws error on /developers. (see comment in code)

File: src/scripts/main-nav.js (hunk 1 above)
Comment (hanbyul-here) [reply src/scripts/main-nav.js]: oh you are right, thanks for the comment. will update.

**Comment by migurski** [comment]
Things [seem vertically misaligned in Safari 9](https://precog.mapzen.com/mapzen/styleguide/35339a0d0b232d60082627dfd94b78879eef9530/samples/blog-main-w-new-header.html) and Chrome:

alignment

**Comment by hanbyul-here** [comment]
I forced (put a little style) to be aligned correctly.

sidenote is it rendered correctly on local. checked it is using same styleguide from local, html elements are same, even computed styles are same. really couldn't figure out why  😿

**Comment by migurski** [comment]
That’s weird! This looks much better, though the compass icon is still weirdly positioned.

**Comment by hanbyul-here** [comment]
I think you were talking about vertical alignment of compass component? please check it again (and thanks for catching it :) ) @migurski

**Comment by migurski** [comment]
Everything looks pretty good in https://precog.mapzen.com/mapzen/styleguide/70733d9/samples/blog-main-w-new-header.html

**Comment by souperneon** [comment]
I might sound like a total downer saying this, but the compass although really fun is visually confusing to have. It's kinda a separator but kinda part of the sign in link and it makes me think why just that link is special?
It will be less confusing to users if we make a separator that makes sign in and sign up a section in the nav or get rid of it. Thoughts @migurski @hanbyul-here ?

**Comment by hanbyul-here** [comment]
Compass is visual charm but also it is something visually coherent before and after login. (After login screenshot attached) I thought we want more attention for sign-in/ up? but I am also open to ditch the compass
screen shot 2016-10-25 at 5 41 49 pm

**Comment by hanbyul-here** [comment]
we are dark launching. It will be better not to drag this any more. We can talk more before it becomes real thing visually @souperneon

**Comment by souperneon** [comment]
I see your point @hanbyul-here let's launch it :)

**Pull request merged**